From b1c5f6160aaae4ec2dec75e2e949440e4e4ddfe5 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Thu, 18 Jul 2013 15:00:29 +0800
Subject: fix the stuck issue when click the pause button

BZ: 123690

root cause: the video event thread is scheduled frequently for it is posted
with no delay, and it will hold the mutex lock, so the pause operation can't
get the lock on time.
This patch adds a delay to post the video event(the default value is 10ms), so
any attempts to cancel future video event could take effect within this 10ms.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: I209b48ab7e60e6fc016ebd9c2fcf2aa725c4fc49
Orig-Change-Id: I48c7f02f342517e77c21f92d0fffd6ac6c4caa00
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/lvpp/PreviewPlayer.cpp |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/libvideoeditor/lvpp/PreviewPlayer.cpp b/libvideoeditor/lvpp/PreviewPlayer.cpp
index f547260..bc25090 100755
--- a/libvideoeditor/lvpp/PreviewPlayer.cpp
+++ b/libvideoeditor/lvpp/PreviewPlayer.cpp
@@ -1092,7 +1092,10 @@ void PreviewPlayer::onVideoEvent() {
         }
 
         if(!mIsVideoSourceJpg) {
-            postVideoEvent_l(0);
+            // The next video event scheduling will occur after 10ms so that
+            // any attempts to cancel future video events could take effect within
+            // this 10ms interval
+            postVideoEvent_l();
         }
         else {
             postVideoEvent_l(33000);
-- 
1.7.4.1

